### PR TITLE
AudioBufferSourceNode: inconsistencies in fast and slow track

### PR DESCRIPTION
**Pull request by b-ma**
Work in progress for #525

The failing tests are passing now but it introduced / revealed another issue, `test_loop_hangs` is now failing for another reason....

The semantics of loop start and end are not clear and they are thus not applied properly in edge cases I think, I need / will take some time to review all this properly

### CODE REVIEW

**Comment by orottier** [comment]
Right, I understand. Let me know if I can help somehow!

**Comment by b-ma** [comment]
Ok, I think the failing test was faulty actually, I renamed it to `test_loop_out_of_bounds` and added some comment, could you confirm you read it the same as me?

I left a bunch of todos I will try to tackle in the next days

File: src/node/audio_buffer_source.rs [hunk 1]
```diff
@@ -1461,14 +1590,25 @@ mod tests {
...
 
-        assert_float_eq!(channel[0], 1.0, abs_all <= 0.);
-        assert_float_eq!(channel[1..], [0.; 48_000 - 1][..], abs_all <= 0.);
+        // Both loop points will be clamped to buffer duration due to rules defined at
+        // https://webaudio.github.io/web-audio-api/#dom-audiobuffersourcenode-loopstart
+        // https://webaudio.github.io/web-audio-api/#dom-audiobuffersourcenode-loopend
+        // Thus it violates the rule defined in
+        // https://webaudio.github.io/web-audio-api/#playback-AudioBufferSourceNode
+        // `loopStart >= 0 && loopEnd > 0 && loopStart < loopEnd`
+        // Hence the whole buffer should be looped
```
Comment (orottier) [inline src/node/audio_buffer_source.rs:1604]: Yes, I agree. I did not put too much thought into the expected result earlier, just focused on fixing the indefinite loop. Thanks!

**Comment by b-ma** [comment]
Ok I think it's all good to me for this one

**Comment by orottier** [comment]
Benchmarks are currently failing #529 
I was interested in the performance impact for the interpolation moving to f64. It requires some casting back and forth and increases the size of a 'hot' object. I will run the locally instead. Anyway, since this PR improves correctness, so the results of benchmarks are not too important

**Comment by orottier** [comment]
Okay maybe we don't have benchmarks for slow-track processing anyway.. oh well

**Pull request merged**